python/cuml/pytest_benchmarks/test_ensemble.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmark integrations described in blog:
+   https://medium.com/rapids-ai/100x-faster-machine-learning-model-ensembling-with-rapids-cuml-and-scikit-learn-meta-estimators-d869788ee6b1
+"""
+
+from cuml.test.utils import unit_param, stress_param
+from cuml.benchmark.algorithms import AlgorithmPair
+from cuml.common.import_utils import has_pytest_benchmark
+
+import pytest
+
+from sklearn.datasets import make_regression, make_classification
+from sklearn.ensemble import AdaBoostRegressor, VotingClassifier
+from sklearn.ensemble import StackingClassifier, BaggingRegressor
+from sklearn.linear_model import LogisticRegression
+from sklearn.neighbors import KNeighborsRegressor
+from sklearn.svm import SVR, SVC
+
+import functools
+
+import cuml
+
+#
+# Testing utilities
+#
+
+
+@functools.lru_cache(maxsize=8)
+def gendata(datafunc, n_samples, n_features, **data_kwargs):
+    return datafunc(n_samples=n_samples, n_features=n_features, **data_kwargs)
```

Comment:
If this function needed? Can you use the cuml.benchmarks.gen_data function or expand that? It already includes caching and datasets like "classification" and is meant for this usecase

Reply:
Yes, I did find it necessary. I thought about adding an sklearn generator to gen_data module, but opted for this isolated implementation until a shared dependency necessitated promoting it to the gen_data module.